kv/kvserver: TestReplicaGCQueueDropReplicaDirect failed

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2079492&tab=buildLog) on [master@542c83e7db1694208e09c7fea7d350db4993b897](https://github.com/cockroachdb/cockroach/commits/542c83e7db1694208e09c7fea7d350db4993b897):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect808729660
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect808729660
--- FAIL: TestReplicaGCQueueDropReplicaDirect (1.92s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2082093&tab=buildLog) on [master@8e38c32b75b52c8d552a25c990cfa16f87e3f04d](https://github.com/cockroachdb/cockroach/commits/8e38c32b75b52c8d552a25c990cfa16f87e3f04d):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect766569446
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect766569446
--- FAIL: TestReplicaGCQueueDropReplicaDirect (1.58s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2084047&tab=buildLog) on [master@8b2cc20a39226384a11ce5c8d5cdea3a691593f2](https://github.com/cockroachdb/cockroach/commits/8b2cc20a39226384a11ce5c8d5cdea3a691593f2):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect842299604
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect842299604
--- FAIL: TestReplicaGCQueueDropReplicaDirect (3.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2087561&tab=buildLog) on [master@161d68e8c2149d401738d2c52cf70071a1c3b610](https://github.com/cockroachdb/cockroach/commits/161d68e8c2149d401738d2c52cf70071a1c3b610):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect400486083
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect400486083
--- FAIL: TestReplicaGCQueueDropReplicaDirect (1.82s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2100339&tab=buildLog) on [master@b901074cd8c2b115affb2b8f5dd89d84e5cf6e32](https://github.com/cockroachdb/cockroach/commits/b901074cd8c2b115affb2b8f5dd89d84e5cf6e32):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect958484110
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect958484110
--- FAIL: TestReplicaGCQueueDropReplicaDirect (2.26s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2104476&tab=buildLog) on [master@8354896f7aa141132765c366ae7ddce9e9e7f361](https://github.com/cockroachdb/cockroach/commits/8354896f7aa141132765c366ae7ddce9e9e7f361):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect700870106
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect700870106
--- FAIL: TestReplicaGCQueueDropReplicaDirect (3.24s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2111568&tab=buildLog) on [master@b8a50cc4d062293915969cdc83e3ec4d057cede5](https://github.com/cockroachdb/cockroach/commits/b8a50cc4d062293915969cdc83e3ec4d057cede5):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect945950930
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect945950930
--- FAIL: TestReplicaGCQueueDropReplicaDirect (1.52s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaGCQueueDropReplicaDirect failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2114512&tab=buildLog) on [master@bfa6307c292ef4dfed4a53cb99f506e6dab26533](https://github.com/cockroachdb/cockroach/commits/bfa6307c292ef4dfed4a53cb99f506e6dab26533):

```
=== RUN   TestReplicaGCQueueDropReplicaDirect
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect531698371
    TestReplicaGCQueueDropReplicaDirect: test_log_scope.go:58: use -show-logs to present logs inline
    TestReplicaGCQueueDropReplicaDirect: client_replica_gc_test.go:96: r35 was not found on s2
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaGCQueueDropReplicaDirect531698371
--- FAIL: TestReplicaGCQueueDropReplicaDirect (1.85s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicaGCQueueDropReplicaDirect PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaGCQueueDropReplicaDirect.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/client_replica_gc_test.go
+++ b/pkg/kv/kvserver/client_replica_gc_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
 )
 
 // TestReplicaGCQueueDropReplica verifies that a removed replica is
@@ -72,7 +73,7 @@ func TestReplicaGCQueueDropReplicaDirect(t *testing.T) {
 
 	tc = testcluster.StartTestCluster(t, numStores,
 		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationAuto,
+			ReplicationMode: base.ReplicationManual,
 			ServerArgs: base.TestServerArgs{
 				Knobs: base.TestingKnobs{
 					Store: &testKnobs,
@@ -82,8 +83,11 @@ func TestReplicaGCQueueDropReplicaDirect(t *testing.T) {
 	)
 	defer tc.Stopper().Stop(context.Background())
 
+	// Create our scratch range and up-replicate it.
 	k := tc.ScratchRange(t)
-	desc := tc.LookupRangeOrFatal(t, k)
+	tc.AddReplicasOrFatal(t, k, tc.Target(1), tc.Target(2))
+	require.NoError(t, tc.WaitForVoters(k, tc.Target(1), tc.Target(2)))
+
 	ts := tc.Servers[1]
 	store, pErr := ts.Stores().GetStore(ts.GetFirstStoreID())
 	if pErr != nil {
@@ -91,10 +95,9 @@ func TestReplicaGCQueueDropReplicaDirect(t *testing.T) {
 	}
 
 	{
-		repl1, err := store.GetReplica(desc.RangeID)
-		if err != nil {
-			t.Fatal(err)
-		}
+		repl1 := store.LookupReplica(roachpb.RKey(k))
+		require.NotNil(t, repl1)
+
 		eng := store.Engine()
 
 		// Put some bogus sideloaded data on the replica which we're about to
@@ -122,7 +125,8 @@ func TestReplicaGCQueueDropReplicaDirect(t *testing.T) {
 					repl1.RaftLock()
 					dir := repl1.SideloadedRaftMuLocked().Dir()
 					repl1.RaftUnlock()
-					if _, err := eng.Stat(dir); os.IsNotExist(err) {
+					_, err := eng.Stat(dir)
+					if os.IsNotExist(err) {
 						return nil
 					}
 					return errors.Errorf("replica still has sideloaded files despite GC: %v", err)
@@ -131,12 +135,12 @@ func TestReplicaGCQueueDropReplicaDirect(t *testing.T) {
 		}()
 	}
 
-	desc = tc.RemoveReplicasOrFatal(t, k, tc.Target(1))
+	desc := tc.RemoveReplicasOrFatal(t, k, tc.Target(1))
 
 	// Make sure the range is removed from the store.
 	testutils.SucceedsSoon(t, func() error {
 		if _, err := store.GetReplica(desc.RangeID); !testutils.IsError(err, "r[0-9]+ was not found") {
-			return errors.Errorf("expected range removal: %v", err) // NB: errors.Wrapf(nil, ...) returns nil.
+			return errors.Errorf("expected range removal: %v", err)
 		}
 		return nil
 	})

